enhancements/sig-architecture/10-deletepropagationstrategy/README.md

```diff
@@ -0,0 +1,205 @@
+# Introduce DeletePropagation in ManifestWork
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management/website/)
+
+## Summary
+
+Today, when a manifestwork is deleted or a manifest resource in the manifestwork is removed, the work agent
+deletes the related resources on the managedcluster in the `ForegroundDeletion` strategy. It means the manifestwork
+will be kept in deleting status until the related resource owned by the manifestwork is deleted on the managedcluster.
+There is use cases that asks for othere deletion strategies. This proposal is to introduce a property in manifestwork
+that the user can define the orphan deletion propagation for resources in the manifestwork.
+
+## Motivation
+
+### Goals
+
+- Define the DeletePropagation property in manifestwork
+- Indicate the behavior of work agent upon the existence of DeletePropagation property
+
+### Non-Goals
+
+- Deletion strategy other than orphan.
+
+## Proposal
+
+We will introduce a `DeletePropagations` property in the manifestwork API.
+
+### User Stories
+
+#### Story 1
+
+As a manifestwork user, I create a manifeswork with a list of resources. When I delete the manifestwork on the hub,
+I do not want the resources defined in the manifestwork to be garbage collected on the managedcluster.
+
+#### Story 2
+
+As a manifestwork user, I create a manifeswork with a list of resources. When I remove one or several resources from
+the manifestwork, I do not what these resources to be garbage collected on the managedcluster.
+
+
+## Design Details
+
+### Manifestwork API update
+
+We will introduce a `DeletePropagations` property as following:
+
+```go
+// ManifestWorkSpec represents a desired configuration of manifests to be deployed on the managed cluster.
+type ManifestWorkSpec struct {
+	// Workload represents the manifest workload to be deployed on a managed cluster.
+	Workload ManifestsTemplate `json:"workload,omitempty"`
+
+	// DeletePropagationOrphan represents resources to be orphaned when the manifestwork is deleted.
+	// Foreground deletion strategy is applied to all the resource in this manifestwork when it is nil.
+  // Orphan deletion strategy is applied to all the resource in this manifestwork when it is empty.
+	// +optional
+	DeletePropagationOrphan *DeletePropagationStrategy `json:"deletePropagationsOrphan,omitempty"`
+}
+
+type DeletePropagationStrategy struct {
```

Comment:
rules like this work best you can specify a slice of them.   Is this API close to what you want to describe?

```go
type DeletePropagationStrategy struct{
    // type can be Orphan or SelectivelyOrphan
    // SelectivelyOrphan should be rarely used.  It is provided for cases where a particular resource is transfering
    // ownership from one ManifestWork to another or another management unit.
    // Setting this value will allow a flow like
    // 1. create manifestwork/2 to manage foo
    // 2. update manifestwork/1 to selectively orphan foo
    // 3. remove foo from manifestwork/1 without impacting continuity because manifestwork/2 adopts it.
    Type string

    SelectivelyOrphan *SelectivelyOrphan
}

type SelectivelyOrphan struct{
    // each orphaningrule identifies a single resource included in this manifestwork
    OrphaningRules []OrphaningRule
}

type OrphaningRule struct{
  Group string
  Resource string
  ResourceNamespace string
  ResourceName string
}
```

Reply:
That looks great, updated.